File: src/Customer.ts
import { EmailSender } from './EmailSender.js'

export class Customer {
  private readonly name: string
  private readonly email: string
  private readonly birthday: Date

  constructor(name: string, email: string, birthday: Date) {
    this.name = name
    this.email = email
    this.birthday = birthday
  }

  hasBirthday(aDate: Date): boolean {
    return this.birthday == aDate
  }

  sendEmail(template: string, sender: EmailSender) {
    const message = template.replace('{name}', this.name)
    sender.send(this.email, message)
  }

  fillWithEmail(template: string) {
    return template.replace('{email}', this.email)
  }
}

File: src/main/java/Customer.java
package main.java;

import java.util.Date;

public class Customer {
    private final String name;
    private final String email;
    private final Date birthday;

    public Customer(String name, String email, Date birthday) {
        this.name = name;
        this.email = email;
        this.birthday = birthday;
    }

    public boolean hasBirthday(Date aDate) {
        return birthday.equals(aDate);
    }

    public void sendEmail(String template, EmailSender sender) {
        String message = template.replace("{name}", name);
        sender.send(email, message);
    }

    public String fillWithEmail(String template) {
        return template.replace("{email}", email);
    }
} 